devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/tooling/ToolingUtils.java

```diff
@@ -38,21 +43,100 @@ public static boolean isEnforcedPlatform(ModuleDependency module) {
                 || Category.REGULAR_PLATFORM.equals(category.getName()));
     }
 
-    public static IncludedBuild includedBuild(final Project project,
-            final ProjectComponentIdentifier projectComponentIdentifier) {
-        final String name = projectComponentIdentifier.getBuild().getName();
+    public static IncludedBuild includedBuild(final Project project, final String buildName) {
+        Gradle currentGradle = project.getRootProject().getGradle();
+        while (null != currentGradle) {
+            for (IncludedBuild ib : currentGradle.getIncludedBuilds()) {
+                if (ib instanceof IncludedRootBuild) {
+                    continue;
+                }
+
+                if (ib.getName().equals(buildName)) {
+                    return ib;
+                }
+            }
+
+            currentGradle = currentGradle.getParent();
+        }
+
+        return null;
+    }
+
+    public static Project includedBuildProject(IncludedBuildInternal includedBuild, final String projectPath) {
+        return includedBuild.getTarget().getMutableModel().getRootProject().findProject(projectPath);
+    }
+
+    public static Project findLocalProject(final Project project, final String projectPath) {
+        if (projectPath.startsWith(":")) {
+            return project.getRootProject().findProject(projectPath);
+        } else {
+            Project currentProject = project;
+            while (currentProject != null) {
+                final Project foundProject = currentProject.findProject(projectPath);
+                if (foundProject != null) {
+                    return foundProject;
+                }
+
+                currentProject = currentProject.getParent();
+            }
+
+            return null;
+        }
+    }
+
+    public static Project findLocalProject(final Project project, final ArtifactCoords artifactCoords) {
+        for (Project subproject : project.getRootProject().getSubprojects()) {
+            if (subproject.getGroup().equals(artifactCoords.getGroupId()) &&
+                    subproject.getName().equals(artifactCoords.getArtifactId()) &&
+                    (artifactCoords.getVersion() == null || subproject.getVersion().equals(artifactCoords.getVersion()))) {
```

Comment:
Nit: Object.equals ?

Reply:
Gradle projects can all have separate versions, so I want to only explicitly use the version for matching the projects when it is provided in the artifact coordinates. If there is no specified version it should just use the local project as a dependency, if there is a version it should only use the local project if the version matches. Otherwise it would refer to a remote artifact (e.g. Maven dependency)

Reply:
NVM - I had `Object.equals(artifactCoords.getVersion(), subproject.getVersion())` in mind - but that doesn't cover the `null` special case